quickwit-search/src/lib.rs

```diff
@@ -205,6 +205,10 @@ pub async fn single_node_search(
     .map_err(|err| {
         SearchError::InternalError(format!("Failed to build doc mapper. Cause: {}", err))
     })?;
+
+    // Try to build query against current schema
+    let _query = doc_mapper.query(doc_mapper.schema(), search_request)?;
+
```

Comment:
`single_node_search` now builds the query, so in cases where this function is used to search, if the query is invalid, it returns an error instead of returning a result with 0 hits. If this is something we don't want, I can revert the changes.

Reply:
Yes, `single_node_search` is only used in CLI. I added the request validation as well.